server/icepeak-server.nix

```diff
@@ -13,16 +13,21 @@ let
     p.requests
   ]);
   libs = with pkgs; [ ];
+  haskell = pkgs.haskellPackages;
 in
   rec {
-    icepeak-server = pkgs.haskellPackages.callPackage ./server.nix { };
-    stack = pkgs.haskellPackages.stack;
+    icepeak-server = haskell.callPackage ./server.nix { };
+    stack = pkgs.stack;
     icepeak-server-env = pkgs.buildEnv {
       name = "icepeak-server-env";
       paths = [
-        stack
+        haskell.stack
```

Comment:
Do we need to get stack specifically from the Haskell package set? The global one (`pkgs.stack`) might be in the caches, whereas the Haskell package might need to be built.